lisa/tools/git.py

```diff
@@ -44,6 +44,7 @@ def clone(
         url: str,
         cwd: pathlib.PurePath,
         ref: str = "",
+        branch: str = "",
```

Comment:
The branch is one of the ref format, what's the error now?

Reply:
The issue is when checking out with branch name passed as `ref`. 
`legacy_runner` wants to switch to v2 branch provided in runbook but the checkout method called inside clone doesnt translate to actual git checkout command.  Instead it translates to 
`git checkout <branch_name passed as ref> -b  <PurePath>` which fails trying to search for commit id with <branch_name>. Instead it should be a simple `git checkout <branch_name>`. Hence, added branch parameter, so that whenever only branch without any ref is provided, it simply uses git checkout.

Reply:
Could you try again? It should be supported. The `-b <local branch name>` should be the local branch name, not related to which remote branch is checked out. If it's not specified, the remote branch name will be used. I have a try on my local, it works well with branch too. Would you show me the exact errors you found?

Reply:
```
Traceback (most recent call last):
  File "C:\agent\_work\18\s\lisa\lisa\main.py", line 89, in <module>
    exit_code = main()
  File "C:\agent\_work\18\s\lisa\lisa\main.py", line 78, in main
    exit_code = args.func(args)
  File "C:\agent\_work\18\s\lisa\lisa\commands.py", line 48, in run
    raise identifier
  File "C:\agent\_work\18\s\lisa\lisa\commands.py", line 44, in run
    asyncio.run(runner.start())
  File "C:\Python38\lib\asyncio\runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "C:\Python38\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "C:\agent\_work\18\s\lisa\lisa\runner.py", line 124, in start
    raise identifer
  File "C:\agent\_work\18\s\lisa\lisa\runner.py", line 121, in start
    self._start_loop()
  File "C:\agent\_work\18\s\lisa\lisa\runner.py", line 273, in _start_loop
    task = runner.fetch_task()
  File "C:\agent\_work\18\s\lisa\lisa\runners\legacy_runner.py", line 99, in fetch_task
    git.clone(
  File "C:\agent\_work\18\s\lisa\lisa\tools\git.py", line 73, in clone
    self.checkout(ref, cwd=full_path)
  File "C:\agent\_work\18\s\lisa\lisa\tools\git.py", line 91, in checkout
    result.assert_exit_code(message=f"failed to checkout branch. {result.stdout}")
  File "C:\agent\_work\18\s\lisa\lisa\util\process.py", line 37, in assert_exit_code
    return assert_that(self.exit_code, message).is_equal_to(expected_exit_code)
```

This is the stack trace